### PR TITLE
[pfcwd]: add flag to enable pfc wd on start

### PR DESCRIPTION
**Pull request by sihuihan88**
Signed-off-by: Sihui Han <sihan@microsoft.com>

**- What I did**
Add flag to enable pfc wd on start
**- How I did it**
Add the option to enable pfc wd on start
**- How to verify it**

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -8,6 +8,7 @@
 #  * ENABLE_DHCP_GRAPH_SERVICE: Enables get-graph service to fetch minigraph files
 #    through http.
 #  * SHUTDOWN_BGP_ON_START: Sets admin-down state for all bgp peerings after restart.
+#  * ENABLE_PFC_WD_ON_START: Enable PFCWD on server-facing ports when starts.
```
Comment (lguohan) [inline Makefile]: use PFCWD consistently throughout the CR?

Comment (lguohan) [reply Makefile]: Enable PFC WatchDog (PFCWD) on server-facing ports by default for TOR switch.

File: files/build_templates/sonic_debian_extension.j2 [hunk 2]
```diff
@@ -168,6 +168,10 @@ sudo bash -c "echo enabled=false > $FILESYSTEM_ROOT/etc/sonic/updategraph.conf"
 {% if shutdown_bgp_on_start == "y" %}
 sudo bash -c "echo '{ \"DEVICE_METADATA\": { \"localhost\": { \"default_bgp_status\": \"down\" } } }' >> $FILESYSTEM_ROOT/etc/sonic/init_cfg.json"
 {% endif %}
+{% if enable_pfc_wd_on_start == "y" %}
+sudo bash -c "echo '{ \"DEVICE_METADATA\": { \"localhost\": { \"default_pfc_wd_status\": \"enable\" } } }' >> $FILESYSTEM_ROOT/etc/sonic/init_cfg.json"
```
Comment (lguohan) [inline files/build_templates/sonic_debian_extension.j2]: which agent is reading this DEVICE_METADATA default_pfc_wd_status, where is that code?

File: Makefile (hunk 1 above)
Comment (jleveque) [inline Makefile]: Suggest change to `ENABLE_PFC_WD_ON_START: Enable PFC Watchdog on server-facing ports when SONiC starts.` for clarity.

File: rules/config [hunk 3]
```diff
@@ -36,6 +36,10 @@ DEFAULT_PASSWORD = YourPaSsWoRd
 # bgp service starts.
 # SHUTDOWN_BGP_ON_START = y
 
+# ENABLE_PFC_WD_ON_START - if set to y PFC WD will be enabled all server-facing ports when
```
Comment (jleveque) [inline rules/config]: Suggest changing `WD` to `Watchdog` for clarity.

File: files/build_templates/sonic_debian_extension.j2 (hunk 2 above)
Comment (sihuihan88) [reply files/build_templates/sonic_debian_extension.j2]: https://github.com/Azure/sonic-utilities/pull/213

File: files/build_templates/sonic_debian_extension.j2 [hunk 4]
```diff
@@ -168,6 +168,10 @@ sudo bash -c "echo enabled=false > $FILESYSTEM_ROOT/etc/sonic/updategraph.conf"
 {% if shutdown_bgp_on_start == "y" %}
 sudo bash -c "echo '{ \"DEVICE_METADATA\": { \"localhost\": { \"default_bgp_status\": \"down\" } } }' >> $FILESYSTEM_ROOT/etc/sonic/init_cfg.json"
 {% endif %}
+{% if enable_pfcwd_on_start == "y" %}
+sudo bash -c "echo '{ \"DEVICE_METADATA\": { \"localhost\": { \"default_pfc_wd_status\": \"enable\" } } }' >> $FILESYSTEM_ROOT/etc/sonic/init_cfg.json"
```
Comment (lguohan) [inline files/build_templates/sonic_debian_extension.j2]: to be consistent, this should also be default_pfcwd_status

File: files/build_templates/sonic_debian_extension.j2 (hunk 2 above)
Comment (lguohan) [reply files/build_templates/sonic_debian_extension.j2]: if the PR is related/depended to other PR, can you mark this information in the PR comments?

File: Makefile [hunk 5]
```diff
@@ -63,7 +64,7 @@ SONIC_BUILD_INSTRUCTION :=  make \
                            BUILD_NUMBER=$(BUILD_NUMBER) \
                            ENABLE_DHCP_GRAPH_SERVICE=$(ENABLE_DHCP_GRAPH_SERVICE) \
                            SHUTDOWN_BGP_ON_START=$(SHUTDOWN_BGP_ON_START) \
-                           ENABLE_PFC_WD_ON_START=$(ENABLE_PFC_WD_ON_START) \
+                           ENABLE_PFCWD_ON_START=$(ENABLE_PFCWD_ON_START) \
```
[the remark this answers is not in the document]
Comment (lguohan) [reply Makefile]: looks like this issue.

https://github.com/Azure/sonic-buildimage/issues/1201

looks like joe has some suggestion to fix this problem.

**Pull request merged**